### ISSUE TITLE
Preview issues

### ISSUE THREAD

**Issue by hlj24** [thegetty/quire-docs#203]
Hi, Friends,

I've been working on an initial preview for a project to show folks how content would look in Quire. But I can't get it to display locally for viewing while I'm drafting, and when I try on Github, so that I can send to colleagues, I get an unstyled version of quire-starter without any of the files I've added.

I'm running npm 6.14.5, node 12.18.0, and quire 0.18.0. Fun complexity: I run these updates and other admin tasks as root user in order to get the write permissions needed. I usually try to run quire preview and deploy to Github as my own managed account, which only has some admin privileges. Another fun complexity: I had an older version of quire installed, and I initially had to go through the update documentation a few times to get quire to (what I think) is the latest correctly installed version.

For local preview: I have localhost running on my Mac in Mojave, but it's unable to load `localhost:1313` or the alternative written in the documentation following `quire preview`.

For Github: I have a private repo set up. I push changes from the Desktop. I have github.yml configured with the `baseURL` as `https://dukewired.github.io/tenyr-test/` and `canonifyURLs:true`. Then I run `bin/deploy.sh`.

I feel like I'm missing two very obvious things, but I'm not seeing. Can you please give me a hint?

Thanks!

Hannah

**Comment by geealbers** [comment]
Hi Hannah,

For the GitHub site, there's a bug in the latest version of Quire that's causing things not to resolve properly, especially for GH deploys. Apologies for that. We'll hopefully be able to get it fixed in the next release, but in the meantime, in `config/environments/github.yml`, use these settings:

```
baseURL: "https://dukewired.github.io/tenyr-test/"
relativeURLs: true
canonifyURLs: true
```

And in `themes/quire-starter-theme/webpack/webpack.config.prod.js`, change line 80 to:

```
outputPath: "tenyr-test/img/"
```

And line 92 to:

```
outputPath: "tenyr-test/fonts/"
```

For your local previewing issue, could you run `quire preview --verbose` and look for any ERROR messages? My hunch is that it's maybe a YAML validation issue or something that could be hanging up the preview. But if you don't see any errors listed, or you do but don't know what they mean, can you copy the output of your Terminal and share it here?

**Comment by hlj24** [comment]
Hi, Greg,

Thanks for this. I updated the code for the Github preview as you recommended, but I'm still seeing the plain HTML with the starter content.

Here are the errors I'm getting for the local preview:

```
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: spawn /Users/[username]/quire-cli/node_modules/hugo-bin/vendor/hugo ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn /Users/[username]/quire-cli/node_modules/hugo-bin/vendor/hugo',
  path: '/Users/[username]/quire-cli/node_modules/hugo-bin/vendor/hugo',
  spawnargs: [
    'server',
    '--config=config.yml,config/environments/dev.yml',
    '--watch'
  ]
}
```
Thanks!

Hannah

**Comment by geealbers** [comment]
Hm, I'm not 100% sure, but let's try these standard troubleshooting steps that are worth going through as a first step:

- Verify that you're running `quire preview --verbose` or `quire preview` from inside your project folder (which would be whatever you called it when you ran `quire new <projectName>`, presumably `tenyr-test`)
- Verify that running `quire --version` returns  something. Should be `v0.18.0` in your case.
- Reinstall the project node dependencies by deleting the `themes/quire-starter-theme/node_modules` folder, and then running `quire install` inside your project folder. After this process finishes, try `quire preview --verbose` again.

I'm hopeful for that last step in particular, but if none of it gets you anywhere, from your Terminal window please hit Command-S, save the full output to a text file, attach it here and we'll try something else.

**Comment by hlj24** [comment]
-     Verify that you're running quire preview --verbose or quire preview from inside your project folder (which would be whatever you called it when you ran quire new <projectName>, presumably tenyr-test)

**Verfied.**

-     Verify that running quire --version returns something. Should be v0.18.0 in your case.

**Verified.**

-     Reinstall the project node dependencies by deleting the themes/quire-starter-theme/node_modules folder, and then running quire install inside your project folder. After this process finishes, try quire preview --verbose again.

**Attempted. 
[tenyr-test_previewerrors_20200615.txt](https://github.com/gettypubs/quire/files/4782702/tenyr-test_previewerrors_20200615.txt)
**

**Comment by geealbers** [comment]
The `quire install` process looks to have gone okay.. Can you run `quire preview --verbose` and again Command-S the Terminal output and post it here? The `--verbose` flag may/should give us some more information.

**Comment by hlj24** [comment]
Attached: 
[tenyr-test_previewerrors_20200615_verbose.txt](https://github.com/gettypubs/quire/files/4782791/tenyr-test_previewerrors_20200615_verbose.txt)

**Comment by geealbers** [comment]
Let's try installing Hugo on its own. It's supposed to be done as part of the Quire CLI process, but perhaps something went wrong there. Can you run `npm install --global hugo-bin`? The output should look something like the below. And then if it's successful, try previewing again. But if `quire preview --verbose` doesn't work, try `hugo server` and see if that works instead and we can narrow down the problem a bit more.

```
GT26382:~ workstation$ npm install --global hugo-bin
/usr/local/bin/hugo -> /usr/local/lib/node_modules/hugo-bin/cli.js

> hugo-bin@0.60.0 postinstall /usr/local/lib/node_modules/hugo-bin
> rimraf vendor && node lib/install

✔  success   Hugo binary installed successfully!
+ hugo-bin@0.60.0
added 188 packages from 56 contributors in 92.505s
GT26382:~ workstation$ hugo version
Hugo Static Site Generator v0.72.0-8A7EF3CF darwin/amd64 BuildDate: 2020-05-31T12:07:44Z
```

**Comment by hlj24** [comment]
Thanks, Greg. I ran `npm install --global hugo bin` as root because I received permissions errors as my usual account. Here's the output:
```
/usr/local/bin/hugo -> /usr/local/lib/node_modules/hugo-bin/cli.js

> hugo-bin@0.60.0 postinstall /usr/local/lib/node_modules/hugo-bin
> rimraf vendor && node lib/install

✖  fatal     EACCES: permission denied, mkdir '/usr/local/lib/node_modules/hugo-bin/vendor'
Hugo binary installation failed!
+ hugo-bin@0.60.0
added 188 packages from 56 contributors in 12.942s
```
I'm wondering if I need to get my desktop admin involved to see what's blocking my access. Do you have other thoughts?

**Comment by geealbers** [comment]
Yes, it's making the `/usr/local/lib/node_modules/hugo-bin` directory but then not giving you permissions to actually do anything in it. I think you're right that this is a question for your admin. Or you might try manually adjusting the permissions to that folder and retrying the install? If the install finds that folder already there, it might just leave it as is and continue with the process rather than deleting it and starting over (in which case you'd end up with the same permissions issues). You could also try running the install as `sudo npm install` but that's not a thing I can really recommend. 

Sorry for the trouble!

**Comment by hlj24** [comment]
Hi, Greg. I've been working through this with my department's desktop support person (Brad). We could not solve the issue on my computer, which Brad initially thought might be related to how users are grouped on our institution's computers, so we set up a VM and fresh install of a Mac OS. I tried doing a fresh install of Quire just now. I reached step 5 in the Mac installation (`sudo chown -R $USER /usr/local`) and received an "operation not permitted" error. I tried using `sudo chmod -R 777` and the Get Info window to change permissions on the usr directory. Even as a local admin on a fresh OS install, I could not override these permissions. I'm thinking this may be the root cause of my issues above. I'm running Mac OS 10.14.6 Mojave on both the VM and desktop. Have you run into any issues along these lines with Mojave?
